### PR TITLE
Rename hostname to be host and fix use of hostname parm in API

### PR DESCRIPTION
**Pull request by robbrockbank**
Allow host to be passed in on assign ip methods.

### CODE REVIEW

**Comment by robbrockbank** [comment]
Fixes #57

File: calico_containers/pycalico/ipam.py [hunk 1]
```diff
@@ -818,62 +819,3 @@ def get_assignment_attributes(self, address):
         else:
             _, attributes = block.get_attributes_for_ip(address)
             return attributes
-
```
Comment (tomdee) [inline calico_containers/pycalico/ipam.py:821]: We;ll have to make sure this is mentioned in the release note.

**Comment by tomdee** [comment]
Can you confirm that other than the variable renaming and test updates, the only changes are
- add host to auto_assign (+ update the affinity check)
- add host to _auto_assign_block (+ pass through)
- remove a couple of depracated methods?

**Comment by robbrockbank** [comment]
Yes that's it + curresponding text reworking to talk about host ID rather than hostname

**Comment by tomdee** [comment]
OK, so if a user consistently passes in their own host parameter, they could use whatever values they want and get_hostname() will never be called?

If so, LGTM.

**Comment by robbrockbank** [comment]
@tomdee, yes that's correct.

Merging.

File: calico_containers/pycalico/ipam.py (hunk 1 above)
Comment (robbrockbank) [inline calico_containers/pycalico/ipam.py:821]: Yup - will do

**Pull request merged**